x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/response/HuggingFaceRerankResponseEntity.java

```diff
@@ -0,0 +1,130 @@
+    private static final String INVALID_ID_FIELD_FORMAT_TEMPLATE = "Expected numeric value for record ID field in Hugging Face rerank ";
+
+    /**
+     * Parses the Hugging Face rerank response.
+
+     * For a request like:
+     *
+     * <pre>
+     *     <code>
+     *         {
+     *              "input": ["luke", "like", "leia", "chewy","r2d2", "star", "wars"],
+     *              "query": "star wars main character",
+     *              "return_documents": true,
+     *              "top_n": 1
+     *          }
+     *     </code>
+     * </pre>
+
+     * The response would look like:
+
+     * <pre>
+     *     <code>
+     *         {
+     *              "rerank": [
+     *                  {
+     *                       "index": 5,
+     *                       "relevance_score": -0.06920313,
+     *                       "text": "star"
+     *                   }
+     *               ]
+     *          }
+     *     </code>
+     * </pre>
+     */
+
+    public static RankedDocsResults fromResponse(HuggingFaceRerankRequest request, HttpResult response) throws IOException {
+        var parserConfig = XContentParserConfiguration.EMPTY.withDeprecationHandler(LoggingDeprecationHandler.INSTANCE);
+
+        try (XContentParser jsonParser = XContentFactory.xContent(XContentType.JSON).createParser(parserConfig, response.body())) {
+            moveToFirstToken(jsonParser);
+
+            XContentParser.Token token = jsonParser.currentToken();
+            ensureExpectedToken(XContentParser.Token.START_ARRAY, token, jsonParser);
+
+            var rankedDocs = doParse(jsonParser);
+            var rankedDocsByRelevanceStream = rankedDocs.stream()
+                .sorted(Comparator.comparingDouble(RankedDocsResults.RankedDoc::relevanceScore).reversed());
+            var rankedDocStreamTopN = request.getTopN() == null
+                ? rankedDocsByRelevanceStream
+                : rankedDocsByRelevanceStream.limit(request.getTopN());
+            return new RankedDocsResults(rankedDocStreamTopN.toList());
+        }
+    }
+
+    private static List<RankedDocsResults.RankedDoc> doParse(XContentParser parser) throws IOException {
+        return parseList(parser, (listParser, index) -> {
+            var parsedRankedDoc = HuggingFaceRerankResponseEntity.RankedDocEntry.parse(parser);
+
+            if (parsedRankedDoc.id == null) {
+                throw new IllegalStateException(format(FAILED_TO_FIND_FIELD_TEMPLATE, RankedDocEntry.ID.getPreferredName()));
+            }
+
+            if (parsedRankedDoc.score == null) {
+                throw new IllegalStateException(format(FAILED_TO_FIND_FIELD_TEMPLATE, RankedDocEntry.SCORE.getPreferredName()));
+            }
+
+            try {
+                return new RankedDocsResults.RankedDoc(parsedRankedDoc.id, parsedRankedDoc.score, parsedRankedDoc.text);
+            } catch (NumberFormatException e) {
```

Comment:
I might be missing it but what logic could throw the `NumberFormatException`? Do we need the try/catch?